src/error.rs

```diff
@@ -38,10 +38,18 @@ pub enum Error {
 impl fmt::Display for Error {
     fn fmt(&self, fmt: &mut fmt::Formatter<'_>) -> fmt::Result {
         match self {
-            Error::RpcFailure(RpcStatus { status, details }) => match details {
-                Some(details) => write!(fmt, "RpcFailure: {} {}", status, details),
-                None => write!(fmt, "RpcFailure: {}", status),
-            },
+            Error::RpcFailure(s) => {
+                if s.error_message().is_empty() {
+                    write!(fmt, "RpcFailure: {}", s.error_code())
```

Comment:
I think it would help to have the same format whether there is an error message or not. I would tend towards less punctuation: `RpcFailure {}: {}, s.error_code(), s.error_message()`

Reply:
It will leave some blank spaces in the end of string.

Reply:
I don't know if that is a problem. But mostly I don't see the point of putting the code in brackets.